lib/screens/onboarding/onboarding_state_machine.dart

```diff
@@ -0,0 +1,214 @@
+import 'dart:async';
+
+class _Transition {
+  final Events event;
+  final States targetState;
+  Map<String, dynamic> data;
+
+  _Transition({this.event, this.targetState});
+}
+
+class _State {
+  final States name;
+  final List<_Transition> transitions;
+
+  _State({this.name, this.transitions});
+}
+
+enum Events {
+  foundInviteLink,
+  foundNoLink,
+  foundInviteDetails,
+  foundNoInvite,
+  inviteAccepted,
+  inviteRejected,
+  createAccountCanceled,
+  claimInviteCanceled,
+  importAccountCanceled,
+  chosenImportAccount,
+  chosenClaimInvite,
+  accountCreated,
+  accountImported,
+  inviteCodeClaimed,
+  createAccountRequested,
+  importAccountRequested,
+  createAccountFailed,
+  importAccountFailed,
+}
+
+enum States {
+  checkingInviteLink,
+  processingInviteLink,
+  onboardingMethodChoice,
+  inviteConfirmation,
+  createAccount,
+  importAccount,
+  claimInviteCode,
+  creatingAccount,
+  importingAccount,
+  finishOnboarding,
+}
```

Comment:
wow.... that's a lot of states

Reply:
Yeah, lot of states and events - hope covered all needed onboarding scenarios - or at least it makes feel so :smile: